File: ClaimPortalJava/src/main/java/com/mt/claimportaljava/utils/model/LocationBuilder.java
package com.mt.claimportaljava.utils.model;

import com.mt.claimportaljava.dtos.CreateLocationDto;
import com.mt.claimportaljava.models.Location;

public class LocationBuilder {
    private Location location;

    public LocationBuilder() {
        this.location = new Location();
    }

    public Location build() {
        return this.location;
    }

    public void withDataFromDto(CreateLocationDto createLocationDto) {
        withAddress(createLocationDto.getAddress());
        withPostalCode(createLocationDto.getPostalCode());
        withCity(createLocationDto.getCity());
        withCountry(createLocationDto.getCountry());
    }

    public void withAddress(String address) {
        this.location.setAddress(address);
    }

    public void withPostalCode(String postalCode) {
        this.location.setPostalcode(postalCode);
    }

    public void withCity(String city) {
        this.location.setCity(city);
    }

    public void withCountry(String country) {
        this.location.setCountry(country);
    }
}

File: ClaimSystemPython/src/utils/model/location_builder.py
from src.models.location import Location
from src.schemas.location import GetLocationSchema


def build_location(create_location_schema):
    location = Location(address=create_location_schema.address,
                        postal_code=create_location_schema.postal_code,
                        city=create_location_schema.city,
                        country=create_location_schema.country)
    return location


def build_location_response(location):
    location_response = GetLocationSchema(id=location.id,
                                          address=location.address,
                                          postal_code=location.postal_code,
                                          city=location.city,
                                          country=location.country)
    return location_response